### PR TITLE
[QoS]HeadroomPoolsize test changes for dnx- chassis

### PR DESCRIPTION
**Pull request by ansrajpu-git**
### Description of PR

 HeadroomPoolsize test improvised for  dnx-chassis.
Summary:
Below changes are added -

- Packet type changed from tcp packet to ip packet.
- Added code to dynamically compensate packets
- Improvised on sleep time
- Added check for ingress counter for dnx -chassis after headroom is filled  

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?
the test fails intermittently in test runs. 
#### How did you verify/test it?
Executed qos test cases for  dnx- chassis.

### CODE REVIEW

**Comment by ansrajpu-git** [comment]
@vmittal-msft, Kindly review.

**Comment by ansrajpu-git** [comment]
@neethajohn , @vmittal-msft , Kindly review this PR.

**Comment by ansrajpu-git** [comment]
@vmittal-msft , please review & let me know your comments

**Comment by ansrajpu-git** [comment]
@vmittal-msft , please review & let me know your comments

**Comment by vmittal-msft** [comment]
@ansrajpu-git  please rebase.

**Comment by vmittal-msft** [comment]
@ansrajpu-git  please re-base. Added @kperumalbfn  for review.

File: tests/saitests/py3/sai_qos_tests.py [hunk 1]
```diff
@@ -2854,31 +2866,45 @@ def runTest(self):
...
+                # Prepare IP packet data
+                pkt = construct_ip_pkt(self.pkt_size,
+                                       self.pkt_dst_mac,
+                                       self.src_port_macs[sidx_dscp_pg_tuples[i][0]],
+                                       self.src_port_ips[sidx_dscp_pg_tuples[i][0]],
+                                       self.dst_port_ip,
+                                       sidx_dscp_pg_tuples[i][1],
+                                       self.src_port_vlan[sidx_dscp_pg_tuples[i][0]],
+                                       ecn=self.ecn,
+                                       ttl=64)
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:2882]: How are we setting TOS byte here ?

File: tests/saitests/py3/sai_qos_tests.py [hunk 2]
```diff
@@ -2667,7 +2667,8 @@ def setUp(self):
         self.pgs = [pg + 2 for pg in self.test_params['pgs']]
         self.src_port_ids = self.test_params['src_port_ids']
         self.src_port_ips = self.test_params['src_port_ips']
-        self.platform_asic = self.test_params['platform_asic']
+        self.platform_asic = str(self.test_params['platform_asic'])
```
Comment (vmittal-msft) [inline tests/saitests/py3/sai_qos_tests.py:2670]: @ansrajpu-git is this change verified on T0/T1 ?

**Comment by ansrajpu-git** [comment]
@vmittal-msft , We do not have T0/T1 test setup to verify.

File: tests/saitests/py3/sai_qos_tests.py (hunk 1 above)
Comment (ansrajpu-git) [reply tests/saitests/py3/sai_qos_tests.py:2882]: As per the definition it is set as below 
def construct_ip_pkt(pkt_len, dst_mac, src_mac, src_ip, dst_ip, dscp, src_vlan, **kwargs):
    ecn = kwargs.get('ecn', 1)
    ip_id = kwargs.get('ip_id', None)
    ttl = kwargs.get('ttl', None)
    exp_pkt = kwargs.get('exp_pkt', False)

    **tos = (dscp << 2) | ecn**
    pkt_args = {
        'pktlen': pkt_len,
        'eth_dst': dst_mac,
        'eth_src': src_mac,
        'ip_src': src_ip,
        'ip_dst': dst_ip,
        'ip_tos': tos
    }

**Comment by vmittal-msft** [comment]
please re-base, let me see if i can verify.

**Comment by ansrajpu-git** [comment]
> please re-base, let me see if i can verify.

@vmittal-msft . PR rebased now. Please review

**Comment by vmittal-msft** [comment]
@ansrajpu-git could you verify for T0/T1 ?
@kperumalbfn please review

**Comment by ansrajpu-git** [comment]
These changes are improvement over t2 topo Broadcom dnx-chassis. Since with the PR- https://github.com/sonic-net/sonic-buildimage/pull/18239 for new vsq threshold and dynamic buffer, the existing sonic-mgmt  test case for HeadroomPoolsize is obsolete and skipped for now. Since the update on the testcases for t2 dnx-chassis is no more needed. Hence closing this PR. PR for skipping HeadroomPoolsize  # https://github.com/sonic-net/sonic-mgmt/pull/13513

**Pull request closed without merging**